src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -541,95 +530,103 @@ private async ValueTask<TargetingContext> ResolveTargetingContextAsync(Cancellat
             return context;
         }
 
-        private async ValueTask<VariantDefinition> GetAssignedVariantAsync(FeatureDefinition featureDefinition, TargetingContext context, CancellationToken cancellationToken)
+        private async ValueTask<VariantDefinition> GetAssignedVariantAsync(EvaluationEvent evaluationEvent, TargetingContext context, CancellationToken cancellationToken)
         {
             VariantDefinition variantDefinition = null;
 
             if (context != null)
             {
-                variantDefinition = await AssignVariantAsync(featureDefinition, context, cancellationToken).ConfigureAwait(false);
+                variantDefinition = await AssignVariantAsync(evaluationEvent, context, cancellationToken).ConfigureAwait(false);
             }
 
             if (variantDefinition == null)
             {
-                variantDefinition = featureDefinition.Variants.FirstOrDefault((variant) => variant.Name == featureDefinition.Allocation.DefaultWhenEnabled);
+                variantDefinition = evaluationEvent.FeatureDefinition.Variants.FirstOrDefault((variant) => variant.Name == evaluationEvent.FeatureDefinition.Allocation.DefaultWhenEnabled);
+
+                evaluationEvent.AssignmentReason = AssignmentReason.EnabledDefault;
```

Comment:
If there is no default variant for the enabled feature, should the reason be None or EnabledDefault?

Reply:
I think the only additional information `EnabledDefault` gives is that the feature was enabled, and you can already check if the feature was enabled in the telemetry. I also think `None` makes more sense since there's no variant assigned.

Reply:
Update: 
`None` is only used when there are no variants. As soon as a variant is defined then some sort of assignment happens. Even if we return a null variant it's because we got to a point where we decided that the user should get `DefaultWhenDisabled`, or `DefaultWhenEnabled`, etc...